corehq/apps/data_dictionary/tests/test_models.py

```diff
@@ -12,134 +10,114 @@
 )
 
 
-class TestCaseProperty(TestCase):
-    domain_name = 'test-case-property'
+def test_valid_date():
+    valid_date = '2025-11-02'
+    # check_validity raises exception on invalid date so just
+    # calling it and not getting an exception indicates validity
+    prop = CaseProperty(data_type=CaseProperty.DataType.DATE)
+    prop.check_validity(valid_date)
+
+
+@pytest.mark.parametrize('value', [
+    "2022",
+    "2022-14-03",
+    "Embedded 2025-11-02 in a longer string",
+    "  2025-11-02 ",  # caller is responsible for stripping any surrounding whitespace
+])
+def test_check_invalid_date(value):
+    prop = CaseProperty(data_type=CaseProperty.DataType.DATE)
+    with pytest.raises(exceptions.InvalidDate):
+        prop.check_validity(value)
+
+
+def test_unvalidated_type():
+    # Barcodes are not validated
+    prop = CaseProperty(data_type=CaseProperty.DataType.BARCODE)
+    prop.check_validity('123')
+
+
+@pytest.mark.parametrize('value', ['42', '-10', '3.14', '-3.14'])
+def test_check_valid_number(value):
+    prop = CaseProperty(data_type=CaseProperty.DataType.NUMBER)
+    prop.check_validity(value)
+
+
+@pytest.mark.parametrize('value', ['not a number', '12.34.56'])
+def test_check_invalid_number(value):
+    prop = CaseProperty(data_type=CaseProperty.DataType.NUMBER)
+    with pytest.raises(exceptions.InvalidNumber):
+        prop.check_validity(value)
+
+
+@pytest.mark.parametrize('value', [
+    '42.123 -71.456',
+    '42.123 -71.456 100',
+    '42.123 -71.456 100 5',
+    '-42.123 -71.456 -10 5.5',  # Negative altitude, decimal precision
+    '42123 -71456',  # Note: Validation does not check degrees
```

Comment:
Appreciate the comments in some of these parameterized lists. I can definitely see them helping to understand why particular values were chosen